src/context/TodoContext.tsx

```diff
@@ -0,0 +1,186 @@
+import React, { useEffect, useState } from 'react';
+import { Todo } from '../types/Todo';
+import { editTodo, getTodos, postTodo } from '../api/todos';
+import { ErrorType } from '../types/Errors';
+
+type SelectedFilter = 'all' | 'active' | 'completed';
+
+type TodoContext = {
+  todos: Todo[],
+  setTodos: (todos: Todo[]) => void;
+  error: string | null,
+  handleCloseError: () => void;
+  setNewTodoTitle: (newTodoTitle: string) => void;
+  newTodoTitle: string;
+  setError: (errorName: string) => void;
+  handleSelectFilter: (filterType: SelectedFilter) => void;
+  selectedFilter: SelectedFilter;
+  handleError: (errorName: ErrorType) => void;
+  addTodo: (todo: Omit<Todo, 'id'>) => void;
+  USER_ID: number;
```

Comment:
```suggestion
  userId: number;
```
the uppercased variable name indicates that is it constant variable and since there can be multiple users (hopefully 😄 ) in our app, it should be camelCased, despite the hardcoded value